torch_sim/models/graphpes.py

```diff
@@ -0,0 +1,166 @@
+"""An interface for using arbitrary GraphPESModels in torch_sim.
+
+This module provides a TorchSim wrapper of the GraphPES models for computing
+energies, forces, and stresses of atomistic systems. It serves as a wrapper around
+the graph_pes library, integrating it with the torch_sim framework to enable seamless
+simulation of atomistic systems with machine learning potentials.
+
+The GraphPESWrapper class adapts GraphPESModels to the ModelInterface protocol,
+allowing them to be used within the broader torch_sim simulation framework.
+
+Notes:
+    This implementation requires graph_pes to be installed and accessible.
+    It supports various model configurations through model instances or model paths.
+"""
+
+import typing
+from pathlib import Path
+
+import torch
+
+from torch_sim.models.interface import ModelInterface
+from torch_sim.neighbors import vesin_nl_ts
+from torch_sim.state import SimState, StateDict
+
+
+try:
+    from graph_pes import AtomicGraph, GraphPESModel
+    from graph_pes.atomic_graph import PropertyKey, to_batch
+    from graph_pes.models import load_model
+
+except ImportError:
+    PropertyKey = str
+
+    class GraphPESWrapper(torch.nn.Module, ModelInterface):  # type: ignore[reportRedeclaration]
+        """GraphPESModel wrapper for torch_sim.
+
+        This class is a placeholder for the GraphPESWrapper class.
+        It raises an ImportError if graph_pes is not installed.
+        """
+
+        def __init__(self, *_args: typing.Any, **_kwargs: typing.Any) -> None:  # noqa: D107
+            raise ImportError("graph_pes must be installed to use this model.")
+
+
+def state_to_atomic_graph(state: SimState, cutoff: torch.Tensor) -> AtomicGraph:
+    """Convert a SimState object into an AtomicGraph object.
+
+    Args:
+        state: SimState object containing atomic positions, cell, and atomic numbers
+        cutoff: Cutoff radius for the neighbor list
+
+    Returns:
+        AtomicGraph object representing the batched structures
+    """
+    graphs = []
+
+    for i in range(state.n_batches):
+        batch_mask = state.batch == i
+        R = state.positions[batch_mask]
+        Z = state.atomic_numbers[batch_mask]
+        cell = state.cell[i]
```

Comment:
```suggestion
        cell = state.row_vector_cell[i]
```

Our cell convention is different to ASE/vesin so we actually will need the transpose here.

Reply:
Thanks! Updated.